jack-core/src/com/rapleaf/jack/transaction/TransactorMetricsImpl.java

```diff
@@ -0,0 +1,99 @@
+package com.rapleaf.jack.transaction;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class TransactorMetricsImpl implements TransactorMetrics {
+  int longestQueriesListSize;
+
+
+  LinkedList<Long> longestTimes;
+
+  HashMap<StackTraceElement, Long> longestQueries;
+
+
+  //Contains a map with the *longestQueriesListSize* queries that took the most time to execute and their execution time
```

Comment:
Better to give a brief description of the space-saving algorithm there. This is not a common algorithm.